src/OsiMsk/OsiMskSolverInterface.cpp

```diff
@@ -3960,21 +3959,32 @@ OsiMskSolverInterface::addCols(const int numcols,
   debugMessage("Begin OsiMskSolverInterface::addCols(%d, %p, %p, %p, %p)\n", numcols, (void *)cols, (void *)collb, (void *)colub, (void *)obj);
   #endif
 
-  int i, nz = 0, err = MSK_RES_OK;
-  
-  // For efficiency we put hints on the total future size
-  err = MSK_getmaxnumanz(getLpPtr(),
-                         &nz);
-                     
+  int i,maxnz = 0,nz=0, err = MSK_RES_OK;
+
+  err = MSK_getmaxnumanz(
+                     getLpPtr(),
+                     &maxnz);
   checkMSKerror( err, "MSK_getmaxanz", "addCols" );
 
+  err = MSK_getnumanz(
+                     getLpPtr(),
+                     &nz);
+
+  checkMSKerror( err, "MSK_getnumanz", "addCols" );
+
+  int nzfactor = 1;
+  if (maxnz>0&&nz>0)
+	  {
+	  nzfactor = (nz/maxnz);
```

Comment:
maxnz should always be greater-equal nz, I would believe.
So isn't nzfactor always 0, unless nz==maxnz?

The previous version of the code was keeping the difference maxnumanz-numanz from before and after adding columns constant. I can imagine that this was a bad idea if maxnumanz was initially set to the expected size of the problem to be build and one new uses addCols to actually build the problem. There should be no reason to increase maxnumanz again. So removing that complete logic of increasing maxnumanz in addCols and addRows would make sense to me.

Reply:
You are right sorry for my friday afternoon solution. 
Here's some corrections instead of using a int value using a double value would make more sense.
``double nzfactor = 1;
if (maxnz>0&&nz>0)
	{
	nzfactor=(static_cast<double>(nz)/static_cast<double>(maxnz));
	}
``
Removing the whole maximum non-zero logic would be a better solution as you said. I did some tests and my nzfactor stuff is 14% slower than just removing the whole non zero logic :). So like you I think it will be better to remove the whole thing.

Reply:
Since addCols and addRows just repeatedly call addCol and addRow, respectively, it may be better to make sure that maxnumanz is indeed large enough to hold all columns/rows that are to be added in this round. I've added this in trunk, but only checked that it compiles.